streams/src/test/java/org/apache/kafka/streams/state/internals/CachingSessionStoreTest.java

```diff
@@ -162,6 +164,22 @@ public void shouldFetchCorrectlyAcrossSegments() throws Exception {
     }
 
     @Test
+    public void shouldPutAndMaybeForward() throws Exception {
+        final Windowed<String> a1 = new Windowed<>("a", new SessionWindow(0, 0));
+        final Windowed<String> a2 = new Windowed<>("a", new SessionWindow(0, 1));
+        final MockCacheFlushListener<Windowed<String>, Long> flushListener = new MockCacheFlushListener<>();
```

Comment:
I don't think we need to add the extra class just for this case. We could do something like:
```
final List<Change<Long>> values = new ArrayList<>();
        cachingStore.setFlushListener(new CacheFlushListener<Windowed<String>, Long>() {
            @Override
            public void apply(final Windowed<String> key, final Long newValue, final Long oldValue) {
                values.add(new Change<>(newValue, oldValue));
            }
        });
       ...
      assertThat(values, equalTo(Arrays.asList(new Change<>(1L, null), new Change<>(null, 1L), new Change<>(2L, null))));
```